spring-rabbit/src/main/java/org/springframework/amqp/rabbit/retry/RejectAndDontRequeueRecoverer.java

```diff
@@ -35,14 +38,33 @@
  */
 public class RejectAndDontRequeueRecoverer implements MessageRecoverer {
 
-	protected Log logger = LogFactory.getLog(RejectAndDontRequeueRecoverer.class); // NOSONAR protected
+	private final Supplier<String> messageSupplier;
+
+	protected final Log logger = LogFactory.getLog(getClass()); // NOSONAR protected
+
+	/**
+	 * Construct an instance with the default exception message supplier.
+	 */
+	public RejectAndDontRequeueRecoverer() {
+		this(() -> "Retry Policy Exhausted");
+	}
+
+	/**
+	 * Construct an instance with the provided exception message supplier.
+	 * @param messageSupplier the message supplier.
+	 * @since 2.3.7
+	 */
+	public RejectAndDontRequeueRecoverer(Supplier<String> messageSupplier) {
```

Comment:
Can't we also have a plain String-based ctor?
Just in case for typical consistency with what we have so far in many other exceptions...